provider/bidengine/testdata/usd_pricing_oracle.sh

```diff
@@ -0,0 +1,78 @@
+#!/bin/bash
+
+# These are the variables one can modify to change the USD scale for each resource kind
+CPU_USD_SCALE=0.10
+MEMORY_USD_SCALE=0.02
+STORAGE_USD_SCALE=0.01
+ENDPOINT_USD_SCALE=0.02
+
+# used later for validation
+MAX_INT64=9223372036854775807
+
+# local variables used for calculation
+memory_total=0
+cpu_total=0
+storage_total=0
+endpoint_total=0
+
+# read the JSON in `stdin` into $script_input
+read -r script_input
+
+# iterate over all the groups and calculate total quantity of each resource
+for group in $(jq -c '.[]' <<<"$script_input"); do
+  count=$(jq '.count' <<<"$group")
+
+  memory_quantity=$(jq '.memory' <<<"$group")
+  memory_quantity=$((memory_quantity * count))
+  memory_total=$((memory_total + memory_quantity))
+
+  cpu_quantity=$(jq '.cpu' <<<"$group")
+  cpu_quantity=$((cpu_quantity * count))
+  cpu_total=$((cpu_total + cpu_quantity))
+
+  storage_quantity=$(jq '.storage' <<<"$group")
+  storage_quantity=$((storage_quantity * count))
+  storage_total=$((storage_total + storage_quantity))
+
+  endpoint_quantity=$(jq '.endpoint_quantity' <<<"$group")
+  endpoint_quantity=$((endpoint_quantity * count))
+  endpoint_total=$((endpoint_total + endpoint_quantity))
+done
+
+# calculate the total cost in USD for each resource
+cpu_cost_usd=$(bc -l <<<"${cpu_total}*${CPU_USD_SCALE}")
+memory_cost_usd=$(bc -l <<<"${memory_total}*${MEMORY_USD_SCALE}")
+storage_cost_usd=$(bc -l <<<"${storage_total}*${STORAGE_USD_SCALE}")
+endpoint_cost_usd=$(bc -l <<<"${endpoint_total}*${ENDPOINT_USD_SCALE}")
+
+# validate the USD cost for each resource
+if [ 1 -eq "$(bc <<<"${cpu_cost_usd}<0")" ] || [ 0 -eq "$(bc <<<"${cpu_cost_usd}<=${MAX_INT64}")" ] ||
+  [ 1 -eq "$(bc <<<"${memory_cost_usd}<0")" ] || [ 0 -eq "$(bc <<<"${memory_cost_usd}<=${MAX_INT64}")" ] ||
+  [ 1 -eq "$(bc <<<"${storage_cost_usd}<0")" ] || [ 0 -eq "$(bc <<<"${storage_cost_usd}<=${MAX_INT64}")" ] ||
+  [ 1 -eq "$(bc <<<"${endpoint_cost_usd}<0")" ] || [ 0 -eq "$(bc <<<"${endpoint_cost_usd}<=${MAX_INT64}")" ]; then
+  exit 1
+fi
+
+# finally, calculate the total cost in USD of all resources and validate it
+total_cost_usd=$(bc -l <<<"${cpu_cost_usd}+${memory_cost_usd}+${storage_cost_usd}+${endpoint_cost_usd}")
+if [ 1 -eq "$(bc <<<"${total_cost_usd}<0")" ] || [ 0 -eq "$(bc <<<"${total_cost_usd}<=${MAX_INT64}")" ]; then
+  exit 1
+fi
+
+# call the API and find out the current USD price per AKT
+API_RESPONSE=$(curl -s "https://api.coingecko.com/api/v3/simple/price?ids=akash-network&vs_currencies=usd")
+curl_exit_status=$?
+if [ $curl_exit_status != 0 ]; then
+  exit $curl_exit_status
+fi
+usd_per_akt=$(jq '."akash-network"."usd"' <<<"$API_RESPONSE")
```

Comment:
Since this is just bidding do we want to fail here if the answer for this number is 0?

Reply:
Think we should have a better backup then that, what if the oracle drops completely?

Reply:
Up on line 65, if curl fails the script fails. No bid happens.

Reply:
Done